xc7/yosys/synth.tcl

```diff
@@ -1,5 +1,21 @@
 yosys -import
 
+# Infer 3-state IOBUFs.
```

Comment:
I don't have a problem with any of this, I'm wondering if this should be moved into `synth_xilinx` in upstream yosys?  The changes all seem to be independent of symbiflow-arch-defs?

Reply:
You're probably right. I'll check what steps does the `synth_xilinx` command do regarding IOs and see how I can augment it.

In my opinion the fact that i have to map `$tribuf` to `$_TBUF_` "manually" prior to the `iopadmap` is kind of a workaround. Besides that the `iopadmap` is not behaving properly as it always leaves some dangling wires connected directly to the io pad.

Reply:
Part of the reason I think this change might belong to upstream yosys is they can comment whether what you are doing is the "proper" way.  I am not really equiped to say one way or another.

Reply:
@mkurc-ant I think the correct person to talk to is @mwkmwkmwk who has been doing a lot of work in this area?

Reply:
@mithro Ok, I'll do. Meanwhile the only example of how to use the `iopadmap` I found is that reddit post: https://www.reddit.com/r/yosys/comments/4hwlkx/new_support_for_tristate_buffers_in_iopadmap/